public/app/plugins/datasource/elasticsearch/components/QueryEditor/MetricAggregationsEditor/SettingsEditor/BucketScriptSettingsEditor/state/reducer.test.ts

```diff
@@ -10,15 +10,22 @@ import { reducer } from './reducer';
 
 describe('BucketScript Settings Reducer', () => {
   it('Should correctly add new pipeline variable', () => {
-    const expectedPipelineVar: PipelineVariable = {
+    const firstVar: PipelineVariable = {
       name: 'var1',
       pipelineAgg: '',
     };
 
+    const secondVar: PipelineVariable = {
+      name: 'var2',
+      pipelineAgg: '',
+    };
+
     reducerTester<PipelineVariable[]>()
       .givenReducer(reducer, [])
       .whenActionIsDispatched(addPipelineVariable())
-      .thenStateShouldEqual([expectedPipelineVar]);
+      .thenStateShouldEqual([firstVar])
+      .whenActionIsDispatched(addPipelineVariable())
+      .thenStateShouldEqual([firstVar, secondVar]);
```

Comment:
Should we re-index variables when they are removed? For example the test would pass with the following changes allowing for both vars to get the name of `var2`:

```diff
    reducerTester<PipelineVariable[]>()
      .givenReducer(reducer, [])
      .whenActionIsDispatched(addPipelineVariable())
      .thenStateShouldEqual([firstVar])
      .whenActionIsDispatched(addPipelineVariable())
      .thenStateShouldEqual([firstVar, secondVar])
+     .whenActionIsDispatched(removePipelineVariable(0))
+     .thenStateShouldEqual([secondVar])
+     .whenActionIsDispatched(addPipelineVariable())
+     .thenStateShouldEqual([
+       { name: 'var2', pipelineAgg: '' },
+       { name: 'var2', pipelineAgg: '' },
+     ]);
```